content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,213 @@
+
+</Admonition>
+
+</TopBlock>
+
+## About tracing
+
+When you enable tracing, Prisma Client writes out a trace for each request that it makes. **should we say operation here, rather than request**? Each trace includes one or more [spans](https://www.jaegertracing.io/docs/1.35/architecture/#span), each of which represents one stage in the request life-cycle (**check terminology here**), such as a database statement or command, or serialization and deserialization. Each span includes the round trip connection time from Prisma Client to the database and back.
+
+**Maybe a diagram here to show:**
+
+- Trace (Prisma operation)
+
+  - Span (such as a connection)
+  - Span (such as a database statement or command)
+  - Span (such as deserialization)
+
+  You can send tracing output to the console, or analyze it in any OpenTelemetry-compatible tracing system, such as [Jaeger](https://www.jaegertracing.io/).
+
+**TBA: maybe a screenshot in Jaeger with example**
+
+## Output
+
+Depending on the Prisma operation (**request?**), Prisma Client might output any of the following spans:
+
+- `prisma`: Represents the entire Prisma request, from Prisma Client to the database and back. It contains details such as the model and method called by the Prisma Client.
+- `prisma:query_builder`: Represents how long the [Prisma engines](/concepts/components/prisma-engines) take to generate the SQL queries (**would it be more accurate to say "statement or command" instead of queries here? This is not running rhe command, is it? - it's building the query**). This includes deserialization of the Prisma Client request, validation, and transformation into SQL.
+- `prisma:connection`: Represents how long it took for Prisma Client to get a database connection.
+- `prisma:db_query`: Represents a SQL query that was executed against the database, the SQL statement or command (in the tags), and how long the query took to run.
+- `prisma:itx_runner`: The start of an interactive transaction.
+- `prisma:transaction`: The high-level transaction. **need more detail here - what do we mean by high-level?**
+
+## Considerations and prerequisites
+
+**TBA: Garren to provide some notes on performance considerations. We'll say something like "Note that tracing has a significant effect on performance in the following situations". See notes here - https://github.com/prisma/prisma/issues/13503#event-6977115220**
+
+To use tracing, you must do the following:
+
+1. [Install the appropriate dependencies](#step-1-install-up-to-date-prisma-dependencies).
+1. [Enable the `tracing` feature flag in your Prisma schema file](#step-2-enable-the-feature-flag-in-your-prisma-schema-file).
+1. [Configure OpenTelemetry](#step-3-configure-opentelemetry).
+
+### Step 1. Install up-to-date Prisma dependencies
+
+Use version `4.1.0` or higher of the `prisma`, `@prisma/client`, and '@prisma/instrumentation' npm packages.
```

Comment:
4.2.0 since we keeping it silent for this release.

Reply:
Good point!